packages/portfolio-contract/src/durable-queue.exo.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file Durable queue helper backed by a durable map and index pointers.
+ */
+import { Fail } from '@endo/errors';
+import { M } from '@endo/patterns';
+import type { MapStore, StoreOptions } from '@agoric/store';
+import type { Zone } from '@agoric/zone';
+
+export type DurableQueueState<V> = {
+  readonly entries: MapStore<bigint, V>;
+  head: bigint;
+  tail: bigint;
+};
+
+export type DurableQueue<V> = ReturnType<
+  ReturnType<typeof prepareDurableQueue<V>>
+>;
+
+export const prepareDurableQueue = <V>(
+  zone: Zone,
+  queueName: string,
+  options: StoreOptions = {},
+) => {
+  const collections = zone.detached();
+
+  const indicesShape = M.splitRecord(
+    {
+      head: M.bigint(),
+      tail: M.bigint(),
```

Comment:
They're never <0, right?

```suggestion
      head: M.nat(),
      tail: M.nat(),
```